frame/society/src/lib.rs

```diff
@@ -0,0 +1,1372 @@
+
+			let pos = pos as usize;
+			<Bids<T, I>>::mutate(|b|
+				if pos < b.len() {
+					b[pos].2.check_voucher(&voucher)?;
+					<Vouching<T, I>>::remove(&voucher);
+					let who = b.remove(pos).1;
+					Self::deposit_event(RawEvent::Unvouch(who));
+					Ok(())
+				} else {
+					Err(Error::<T, I>::BadPosition)?
+				}
+			)
+		}
+
+		/// As a member, vote on a candidate.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// Parameters:
+		///
+		/// - `candidate`: The candidate that the member would like to bid on.
+		/// - `approve`: A boolean which says if the candidate should be
+		///              approved (`true`) or rejected (`false`).
+		///
+		/// # <weight>
+		/// - One storage read and O(log M) search to check user is a member.
+		/// - One storage read to lookup candidate index.
+		/// - One storage read and O(C) search to check that user is a candidate.
+		/// - One storage write to add vote to votes.
+		/// - One event.
+		/// # </weight>
+		///
+
+		#[weight = SimpleDispatchInfo::FixedNormal(30_000)]
+		pub fn vote(origin, candidate: <T::Lookup as StaticLookup>::Source, approve: bool) {
+			let voter = ensure_signed(origin)?;
+			ensure!(Self::is_member(&voter), Error::<T, I>::NotMember);
+			let candidate = T::Lookup::lookup(candidate)?;
+			ensure!(Self::is_candidate(&candidate), Error::<T, I>::NotCandidate);
+
+			let vote = if approve { Vote::Approve } else { Vote::Reject };
+			<Votes<T, I>>::insert(&candidate, &voter, vote);
+
+			Self::deposit_event(RawEvent::Vote(candidate, voter, approve));
+		}
+
+		/// As a member, vote on the defender.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// Parameters:
+		///
+		/// - `approve`: A boolean which says if the candidate should be
+		///              approved (`true`) or rejected (`false`).
+		///
+		/// # <weight>
+		/// - One storage read and O(log M) search to check user is a member.
+		/// - One storage write to add vote to votes.
+		/// - One event.
+		/// # </weight>
+		///
+
+		#[weight = SimpleDispatchInfo::FixedNormal(20_000)]
+		pub fn defender_vote(origin, approve: bool) {
+			let voter = ensure_signed(origin)?;
+			ensure!(Self::is_member(&voter), Error::<T, I>::NotMember);
+
+			let vote = if approve { Vote::Approve } else { Vote::Reject };
+			<DefenderVotes<T, I>>::insert(&voter, vote);
+
+			Self::deposit_event(RawEvent::DefenderVote(voter, approve));
+		}
+
+		/// Transfer the first matured payout for the sender and remove it from the records.
+		///
+		/// This extrinsic can be called multiple times to claim all matured payouts.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// # <weight>
+		/// - One storage read and O(log M) search to check signer is a member.
+		/// - One storage read to get all payouts for a member.
+		/// - One storage read to get the current block number.
+		/// - Up to one currency transfer call.
```

Comment:
```suggestion
		/// - One currency transfer call.
```

describe what happens on success. if this call results in an `Ok` return, then it'll have exactly one balance operation.